app/models/invite_code_validator.rb

```diff
@@ -0,0 +1,29 @@
+class InviteCodeValidator < ActiveModel::Validator
+  def validate(user)
```

Comment:
This function could be tweaked a bit for clarity i think.
You fetch the record first. Then hand the code (not the record) to `not_existent?` and then hand an attribute of the record to `count_greater_than_zero?`

I think it would be more consistent to only use the code OR the record in this function so the private functions have a similar interface:
```ruby
if not_existent?(invite_code)
  add_error_to_user("This is not a valid code", user)
elsif used_already?(invite_code)
  add_error_to_user("This code has already been used", user)
end
```
(I put invite_code as a placeholder for either `user.invite_code` or `user_invite_code`. The point is using the same)